crates/counter-agg/src/lib.rs

```diff
@@ -0,0 +1,258 @@
+use time_weighted_average::tspoint::TSPoint;
+use serde::{Deserialize, Serialize};
+use regression::{XYPair, RegressionSummary};
+
+pub mod regression;
+pub mod range;
+mod tests;
+
+#[derive(Debug, PartialEq)]
+pub enum CounterError{
+    OrderError,
+    BoundsInvalid,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq)]
+pub struct CounterSummary {
+    pub first: TSPoint,
+    pub second: TSPoint,
+    pub penultimate: TSPoint,
+    pub last: TSPoint,
+    pub reset_sum: f64,
+    pub num_resets: u64,
+    pub num_changes: u64,
+    pub regress: RegressionSummary,
+    pub bounds: Option<range::I64Range>,
+}
+
+// Note that this can lose fidelity with the timestamp, but it would only lose it in the microseconds, 
+// this is likely okay in most applications. However, if you need better regression analysis at the subsecond level, 
+// you can always subtract a common near value from all your times, then add it back in, the regression analysis will be unchanged.
+// TODO Should this be a method?
+fn ts_to_xy(pt: TSPoint) -> XYPair{
+    XYPair{
+        x: pt.ts as f64,
+        y: pt.val,
+    }
+}
+
+pub fn to_seconds(t: f64)-> f64{
```

Comment:
Not sure if this needs to be public, but if so, perhaps `us_to_seconds` would be a more descriptive name.